pkg/controllers/vdb/init_db.go

```diff
@@ -529,14 +644,48 @@ func (g *GenericDatabaseInitializer) getHadoopConfDir() string {
 // to run with a Kerberos config.  If it doesn't the ctrl.Result will have the
 // requeue bool set.
 func (g *GenericDatabaseInitializer) hasCompatibleVersionForKerberos() ctrl.Result {
-	vinf, ok := g.Vdb.MakeVersionInfo()
 	const DefaultKerberosSupportedVersion = "v11.0.2"
-	if !ok || ok && vinf.IsEqualOrNewer(DefaultKerberosSupportedVersion) {
+	// The '%s' are placeholders for the version extracted from a vdb
+	// and the minimum version you must be on to setup server side Kerberos.
+	eventMsg := "The engine (%s) doesn't have the required change to setup Kerberos in " +
+		"the container.  You must be on version %s or greater"
+	return g.hasCompatibleVersion(DefaultKerberosSupportedVersion, eventMsg)
+}
+
+// hasCompatibleVersionForServerSideEncryption checks whether it has the required engine fix
+// to run with S3 server-side encryption.  If it doesn't the ctrl.Result will have the
+// requeue bool set.
+func (g *GenericDatabaseInitializer) hasCompatibleVersionForServerSideEncryption() ctrl.Result {
+	const DefaultSseSupportedVersion = "v12.0.1"
+	// The '%s' are placeholders for the version extracted from a vdb
+	// and the minimum version you must be on to setup server side encryption.
+	eventMsg := "The engine (%s) doesn't have the required change for server-side encryption. " +
```

Comment:
This message is very similar to the one in `hasCompatibleVersionForKerberos`. Could we put that in the generic function and may just pass in the portion that's different (i.e. "server-side encryption")?